src/DefaultBuilder/src/WebApplication.cs

```diff
@@ -237,4 +241,39 @@ private void Listen(string? url)
         addresses.Clear();
         addresses.Add(url);
     }
+
+    private string DebuggerToString()
+    {
+        return $@"ApplicationName = ""{Environment.ApplicationName}"", IsRunning = {(IsRunning ? "true" : "false")}";
+    }
+
+    // Web app is running if the app has been started and hasn't been stopped.
+    private bool IsRunning => Lifetime.ApplicationStarted.IsCancellationRequested && !Lifetime.ApplicationStopped.IsCancellationRequested;
+
+    internal sealed class WebApplicationDebugView(WebApplication webApplication)
+    {
+        private readonly WebApplication _webApplication = webApplication;
+
+        public IServiceProvider Services => _webApplication.Services;
+        public IConfiguration Configuration => _webApplication.Configuration;
+        public IWebHostEnvironment Environment => _webApplication.Environment;
+        public IHostApplicationLifetime Lifetime => _webApplication.Lifetime;
+        public ILogger Logger => _webApplication.Logger;
```

Comment:
It doesn't have a meaningful debug experience today. But I'd like to improve it in either .NET 8 or later.

For example, a logger can display its name and configured log level.

I want to leave it there today rather than remove and then re-add.

Reply:
Improved logger debugging: https://github.com/dotnet/runtime/pull/87754